guides/TrainingModelsHPC.md

```diff
@@ -0,0 +1,169 @@
+    - replace `main` with `blob/commit-hash` to retrieve a specific commit.
+
+    These bash scripts fetch the crabs package from git, install it in the compute node, and run a training job.
+
+1.  **Edit the training bash script if required**
+
+    Ideally, we won't edit the training scripts much. If we find we do, then we may want to consider moving the relevant parameters to the config file.
+
+    When launching a single or an array job, one may want to edit in the bash script:
+
+    - The `EXPERIMENT_NAME`. This should roughly reflect the reason for running the experiment (for example, include `data_augm` for a data augmentation ablation study). Otherwise we use the name of the dataset (Sep2023).
+    - The `MLFLOW_FOLDER`. By default, we point to the "scratch" folder at `/ceph/zoo/users/sminano/ml-runs-all/ml-runs-scratch` . This folder holds runs that we don't need to keep. For runs we would like to consider for evaluation, we will instead point to the folder at `/ceph/zoo/users/sminano/ml-runs-all/ml-runs`.
+
+    Less frequently, one may need to edit:
+
+    - the `DATASET_DIR`: usually we train on the "Sep2023_labelled" dataset;
+    - the `TRAIN_CONFIG_FILE`: usually we point to the file at `/ceph/zoo/users/sminano/cluster_train_config.yaml` which we can edit. Note that all config parameters are logged in MLflow;
+    - the `GIT_BRANCH`, if we want to use a specific version of the package. Usually we will run the version in `main`.
+
+    Additionally for an array job, one may want to edit the number of jobs in the array (by default set to 3):
+
+    - if the number of jobs in the array is edited, the `LIST_SEEDS` needs to be modified accordingly, otherwise we will get an error when launching the job.
+
+1.  **Edit the config if required**
+
+    The config file the bash scripts point to is located at:
+
+    ```
+    /ceph/zoo/users/sminano/cluster_train_config.yaml
+    ```
+
+    Ideally, we edit the config file more than the bash scripts. All the info in the config file is anyways recorded for each run in its MLflow logs.
+
+> [!CAUTION]
+>
+> If we launch a job and then modify the config file before the job has been able to read it, we may be using an undesired version of the config in our job. To avoid this, it is best to wait until you can verify in MLflow that the job has the expected config parameters, and then edit the file for a new job if needed.
+
+Some parameters we often modify in the config:
+
+    - `num_epochs`
+    - `batch_size_train`
+    - `batch_size_test`
+    - `batch_size_val`
+    - `checkpoint_saving` parameters
+    - `iou_threshold`
+    - `num_workers`
+
+5.  **Run the training job using the SLURM scheduler**
+
+    To launch a job, use the `sbatch` command with the training script you fetched (and maybe edited):
+
+    ```
+    sbatch <path-to-fetched-training-bash-script>
+    ```
+
+1.  **Check the status of the training job**
```

Comment:
```suggestion
6.  **Check the status of the training job**
```

Reply:
yeah sorry, in markdown if you label all items in an enumeration with `1., 1., 1.` it will reformat them as `1., 2., 3.` (with the advantage that you don't have to keep track of them all the time). But when I use alerts (i.e., Caution or Tip blocks), they need to be left-aligned and so that resets the numbering of the enumeration.

I changed it so that numbers are restarted after the tip/caution blocks!